packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input.rs

```diff
@@ -328,94 +420,98 @@ impl StateVarInterface<String> for ValueStateVarInterface {
-            //         dependency_idx: 0,
-            //     },
-            //     DependencyValueUpdateRequest {
-            //         instruction_idx: 2,
-            //         dependency_idx: 0,
-            //     },
-            // ])
+            self.bind_value_to
+                .request_change_value_to(desired_value.clone());
+            self.sync_immediate_value.request_change_value_to(true);
+
+            Ok(vec![
+                DependencyValueUpdateRequest {
+                    instruction_idx: 3,
+                    dependency_idx: 0,
+                },
+                DependencyValueUpdateRequest {
+                    instruction_idx: 2,
+                    dependency_idx: 0,
+                },
+            ])
         }
     }
 }
 
 #[derive(Debug, Default)]
 struct ImmediateValueStateVarInterface {
     essential_value: StateVarReadOnlyViewTyped<String>,
-    sync_values: StateVarReadOnlyViewTyped<bool>,
-    // bind_value_to: StateVarReadOnlyViewTyped<String>,
+    sync_immediate_value: StateVarReadOnlyViewTyped<bool>,
+    bind_value_to: StateVarReadOnlyViewTyped<String>,
+    prefill: StateVarReadOnlyViewTyped<String>,
 }
 
 impl StateVarInterface<String> for ImmediateValueStateVarInterface {
     fn return_dependency_instructions(
         &self,
         _extend_source: Option<&ExtendSource>,
+        _parameters: &StateVarParameters,
     ) -> Vec<DependencyInstruction> {
         vec![
-            DependencyInstruction::Essential {
-                prefill: Some("prefill"),
-            },
+            DependencyInstruction::Essential,
             DependencyInstruction::StateVar {
                 component_idx: None,
                 state_var_name: "syncImmediateValue",
             },
-            // DependencyInstruction::StateVar {
-            //     state_var_name: "bindValueTo",
-            // },
+            DependencyInstruction::StateVar {
+                component_idx: None,
+                state_var_name: "bindValueTo",
+            },
+            DependencyInstruction::StateVar {
+                component_idx: None,
+                state_var_name: "prefill",
+            },
         ]
     }
 
-    fn save_dependencies_for_value_calculation(
-        &mut self,
-        dependencies: &Vec<Vec<Dependency>>,
-    ) -> () {
+    fn save_dependencies_for_value_calculation(&mut self, dependencies: &Vec<Vec<Dependency>>) {
         if let StateVarReadOnlyView::String(essential_value) = &dependencies[0][0].value {
             self.essential_value = essential_value.create_new_read_only_view();
         } else {
             panic!("Got a non-string essential value for immediate value of text input.");
         }
 
-        if let StateVarReadOnlyView::Boolean(sync_values) = &dependencies[1][0].value {
-            self.sync_values = sync_values.create_new_read_only_view();
+        if let StateVarReadOnlyView::Boolean(sync_immediate_value) = &dependencies[1][0].value {
```

Comment:
We need to figure out an alternative interface here, because indexing arrays with literal values puts me on edge.